src/createproject.ts

```diff
@@ -0,0 +1,157 @@
+import * as fs from 'async-file'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { JuliaExecutablesFeature } from './juliaexepath'
+import { registerCommand } from './utils'
+
+export class JuliaNewProjectFeature {
+    constructor(private context: vscode.ExtensionContext, private juliaExecutablesFeature: JuliaExecutablesFeature) {
+        this.context.subscriptions.push(registerCommand('language-julia.createNewProject', () => this.createNewProject()))
+    }
+
+    private async createNewProject() {
+
+        const pkgName = await vscode.window.showInputBox({
+            prompt: 'Please enter the name of the project to create.',
+            validateInput: (input) => {
+                if (input === "")
+                    return 'The project name cannot be empty.'
+                return undefined
+            }
+        })
+        if (!pkgName)
+            return
+
+        const authors = await vscode.window.showInputBox({ prompt: 'Please enter the authors of the project', placeHolder: "Default uses 'github.name' and 'github.email' from the global Git config." })
+        if (authors === undefined)
+            return
+
+        let host = await vscode.window.showQuickPick(['github.com', 'gitlab.com', 'bitbucket.org', 'Other'], { placeHolder: 'The URL to the code hosting service where the project will reside.' })
+        if (host === 'Other') {
+            host = await vscode.window.showInputBox({ prompt: 'Please enter the URL to the code hosting service.' })
+        }
+        if (host === undefined)
+            return
+
+        // TODO: add gihub.user as placeholder
+        const user = await vscode.window.showInputBox({
+            prompt: 'Please enter your GitHub (or other code hosting service) username.',
+            validateInput: (input) => {
+                if (input === "")
+                    return 'Username cannot be empty.'
+                return undefined
+            }
+        })
+        if (user === undefined)
+            return
+
+        let juliaVersion = await vscode.window.showQuickPick(
```

Comment:
Can we have a good default here?

Reply:
1.6? Or the user's current version? If it's left blank PkgTemplates will use whatever version that runs the script. That could be the default.